rdbms/src/main/java/org/spine3/server/storage/jdbc/entity/visibility/VisibilityHandlingStorageQueryFactory.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.jdbc.entity.visibility;
+
+import org.slf4j.Logger;
+import org.spine3.server.entity.Visibility;
+import org.spine3.server.storage.jdbc.entity.visibility.query.CreateVisibilityTableQuery;
+import org.spine3.server.storage.jdbc.entity.visibility.query.InsertAndMarkEntityQuery;
+import org.spine3.server.storage.jdbc.entity.visibility.query.InsertVisibilityQuery;
+import org.spine3.server.storage.jdbc.entity.visibility.query.MarkEntityQuery;
+import org.spine3.server.storage.jdbc.entity.visibility.query.SelectVisibilityQuery;
+import org.spine3.server.storage.jdbc.entity.visibility.query.UpdateVisibilityQuery;
+
+/**
+ * And interface for managing a table for storing the {@linkplain Visibility entity visibility}.
+ *
+ * @param <I> ID type of the entity visibility of which is stored
+ * @author Dmytro Dashenkov.
+ * @see VisibilityQueryFactories
+ */
+public interface VisibilityHandlingStorageQueryFactory<I> {
```

Comment:
It's weird for different reasons:
* according to the description, this is a manager for the visibility-related attributes of the entity;
* however in fact it's less than that: it's just a query factory, which to me does not qualify as a "management" per se; someone else, who executes these queries, is a "manager", as she handles the outcome as well;
* the factory which is declared as an interface should be a special case, not a common thing;
* and the last, but not least: I remember discussing that the visibility attributes should be stored right next to the entity to avoid an extra `JOIN`; this, however, looks like an interface for the special `VisibilityAttributes` table.

Something isn't right.